examples/llm_plus_gnn/g_retriever.py

```diff
@@ -0,0 +1,472 @@
+"""This example implements G-retriever using PyG.
+Original Paper: https://arxiv.org/abs/2402.07630
+“G-Retriever significantly reduces hallucinations
+by 54% compared to the [LLM] baseline“.
+Requirements on top of basic PyG:
+`pip install datasets transformers pcst_fast sentencepiece tqdm accelerate`.
+"""
+import argparse
+import gc
+import math
+import re
+import time
+from os import path
+
+import pandas as pd
+import torch
+from torch.nn.utils import clip_grad_norm_
+from tqdm import tqdm
+
+from torch_geometric import seed_everything
+from torch_geometric.datasets import WebQSPDataset
+from torch_geometric.loader import DataLoader
+from torch_geometric.nn.models import GRetriever
+from torch_geometric.nn.nlp import LLM
+
+
+def detect_hallucinate(pred, label):
```

Comment:
Can you add typing to these functions and maybe a short description on the definition of the hallucination here?

Reply:
sure ooto till aug 27, can circle back then if still needed